examples/esp_idf/hello/pytest/README.md

```diff
@@ -0,0 +1,87 @@
+# Hello Test
+
+This is a simple test that determines whether the DUT is able to
+establish a connection with the Golioth backend and send Hello to it.
+The test requires:
+- An attached ESP board with the hello sample flashed on it
+- The project's api key
+- The name of the serial port to which the board is connected
+- WiFi credentials to be present in a YAML file with the following format:
+
+```yaml
+settings:
+  wifi/ssid: MyWiFiSSID
+  wifi/psk: MyWiFiPassword
+```
+
+- Board fixture installed with the following command:
+```sh
+pip install tests/hil/scripts/pytest-hil
+```
+
+- Latest version of python-golioth-tools installed with the following command: 
+```sh
+pip install git+https://github.com/golioth/python-golioth-tools@v0.5.1
+```
+
+Installing python-golioth-tools enables pytest to use 'device' fixture which
+automatically creates a device in the Golioth console for the duration of the test,
+and deletes it after the test ends. This temporary device shows up on Golioth
+console while the test is running and disappears when the test ends.
+
+The test can be invoked by the following command from the hello/pytest location:
+
+```sh
+pytest -s --rootdir . --port /path/to/serial/port --credentials-file /path/to/credentials/file --board esp-idf --api-key api_key
+```
+
+### Sample output
```

Comment:
I think we can omit the Sample output. The import thing is whether the test passes or not, and it will be burdensome to keep this output up to date. We already have the sample output in the example README that shows what the device is expected to print.